src/toniarts/openkeeper/game/state/GameState.java

```diff
@@ -103,6 +109,11 @@ protected void updateProgress(int progress, int max) {
                             setProgress(0.25f + ((float) progress / max * 0.75f));
                         }
                     }.load(assetManager, kwdFile));
+                    
+                    bulletAppState = new BulletAppState();
+                    this.stateManager.attach(bulletAppState);    
+                    rootNode.attachChild(new ThingLoader().load(bulletAppState, assetManager, kwdFile));

```

Comment:
I'm afraid this crashes the system. OnLoad() is called from another thread. See how the "worldNode" is loaded separately, only attached to the root node later. We still have the scenegraph to be figured out, but for now you could just attach the ThingLoader stuff to the "worldNode" and it'll work.